homework/index.js

```diff
@@ -16,31 +30,97 @@
     xhr.send();
   }
 
-  function createAndAppend(name, parent, options = {}) {
-    const elem = document.createElement(name);
-    parent.appendChild(elem);
-    Object.keys(options).forEach(key => {
-      const value = options[key];
-      if (key === 'text') {
-        elem.textContent = value;
-      } else {
-        elem.setAttribute(key, value);
-      }
+  function renderRepositories(repo, ulRepoInfo) {
+    const url = `https://api.github.com/repos/HackYourFuture/${repo}`;
+    fetchJSON(url, (err, repoInfo) => {
+      ulRepoInfo.innerHTML = '';
+      const li = createAndAppend('li', ulRepoInfo, { id: 'li1' });
+      const table = createAndAppend('table', li);
+      const tbody = createAndAppend('tbody', table);
+
+      const tr1 = createAndAppend('tr', tbody);
```

Comment:
You are repeating yourself four times here: create a `tr` with two `td1 children. Can you make your code DRYer by creating a function, say `addRow`, that you could call like this:

```js
addRow(tbody, labelText, value);
```

For the first row you would need to do something special as that one is different from the rest. For instance:

```js
const firstRow = addRow(tbody, 'Name', '');
createAndAppend('a', firstRow.lastChild, { href: ..., text: ... });
```

Update: while I was typing this you pushed some code that addresses this. Still, you may want to consider the solution for the first row.

Reply:
Thanks for the tip. I have tried to do it so I had created addHref func .Now no need that function.